src/librustc/middle/intrinsicck.rs

```diff
@@ -92,7 +92,7 @@ impl<'a, 'gcx, 'tcx> ExprVisitor<'a, 'gcx, 'tcx> {
                     struct_span_err!(self.infcx.tcx.sess, span, E0591,
                                      "`{}` is zero-sized and can't be transmuted to `{}`",
                                      from, to)
-                        .span_note(span, "cast with `as` to a pointer instead")
```

Comment:
The span here was entirely not helpful as far as I can tell; but I'm open to re-adding it if there's disagreement.